tests/i18n/tests.py

```diff
@@ -713,15 +714,13 @@ def test_false_like_locale_formats(self):
         """
         with patch_formats('fr', THOUSAND_SEPARATOR='', FIRST_DAY_OF_WEEK=0):
             with translation.override('fr'):
-                with self.settings(USE_THOUSAND_SEPARATOR=True, THOUSAND_SEPARATOR='!'):
-                    self.assertEqual('', get_format('THOUSAND_SEPARATOR'))
-                    # Even a second time (after the format has been cached)...
-                    self.assertEqual('', get_format('THOUSAND_SEPARATOR'))
+                self.assertEqual('', get_format('THOUSAND_SEPARATOR'))
+                # Even a second time (after the format has been cached)...
+                self.assertEqual('', get_format('THOUSAND_SEPARATOR'))
```

Comment:
Maybe `patch_formats` can be removed. It mostly seems like a poor man's `override_settings()` and is perhaps obsolete with the signal handler you added?